src/azul/indexer.py

```diff
@@ -19,20 +19,21 @@ class BaseIndexer(ABC):
     """
     def __init__(self, properties: BaseIndexProperties) -> None:
         self.properties = properties
+        self.es_client = ESClientFactory.get()
+        self.metadata_downloader = MetadataDownloader()
```

Comment:
This attribute would go away if you eliminated the class.

Reply:
In general we should aim to minimize state unless additional state considerably increases computational efficiency or readability.